src/Action.hs

```diff
@@ -219,7 +219,9 @@ loginByUser = login . view _Id
 loginByName :: (ActionPersist m, ActionUserHandler m) => UserLogin -> m ()
 loginByName n = do
     Just u <- query $ findUserByLogin n  -- FIXME: handle 'Nothing'
-    loginByUser u
+    if isDeletedUser u
+        then logout >> throwError500 "Unknown user identitifer"
+        else loginByUser u
```

Comment:
We need to check this for `login`, `loginByUser` as well.  (I wonder if there is a better way to implement these three in terms of each other?)